openwisp_monitoring/device/apps.py

```diff
@@ -162,3 +164,23 @@ def register_notifcation_types(self):
                 ),
             },
         )
+
+    @classmethod
+    def connect_config_modified(cls):
+        from openwisp_controller.config.models import Config
+
+        if check_settings.AUTO_CONFIG_CHECK:
+            config_modified.connect(
+                cls.config_modified_receiver,
+                sender=Config,
+                dispatch_uid='config_modified',
+            )
+
+    @classmethod
+    def config_modified_receiver(cls, sender, instance, **kwargs):
+        from ..check.tasks import perform_check
+
+        DeviceData = load_model('device_monitoring', 'DeviceData')
+        device = DeviceData.objects.get(config=instance)
+        check = device.checks.get(name='Configuration Modified')
```

Comment:
never query in the code by name, it's too fragile!

The name right now can also be changed by users so this would stop working in that case.
Use another field, eg: `configuration`.

Moreover, what if the Check has been deleted? Please write a test for this case and ensure it's handled without crashes.